modules/heat_transfer/include/relationshipmanagers/GhostBoundary.h

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// App includes
+#include "AutomaticMortarGeneration.h"
+#include "RelationshipManager.h"
+
+// libMesh includes
+#include "libmesh/mesh_base.h"
+
+using libMesh::boundary_id_type;
+using libMesh::CouplingMatrix;
+using libMesh::Elem;
+using libMesh::GhostingFunctor;
+using libMesh::MeshBase;
+using libMesh::processor_id_type;
+
+class GhostBoundary : public RelationshipManager
```

Comment:
Docstrings

Reply:
Docstrings what?

Reply:
Sorry for the shortness

We usually add a docstring here, and before any significant routine of the class

Reply:
doxygen would be good for the class and for any non-override methods and data members

Reply:
I found only  a handful of examples using `params.setDocString` and not much more with this keyword, as you indicate this is commonly used. Where should I find info on this?

Reply:
`MooseApp` was the first class I opened. It has examples of the doxygen that @GiudGiud is asking for
```c++
/**                                                                                                                       
 * Base class for MOOSE-based applications                                                                                
 *                                                                                                                        
 * This generic class for application provides:                                                                           
 * - parsing command line arguments,                                                                                      
 * - parsing an input file,                                                                                               
 * - executing the simulation                                                                                             
 *                                                                                                                        
 * Each application should register its own objects and register its own special syntax                                   
 */                                                                                                                       
class MooseApp : public ConsoleStreamInterface,                                                                           
                 public PerfGraphInterface,                                                                               
                 public libMesh::ParallelObject,                                                                          
                 public MooseBase                                                                                         
{                                                                                                                         
public:                                                                                                                   
#ifdef LIBTORCH_ENABLED                                                                                                   
  /// Get the device torch is supposed to be running on.                                                                  
  torch::DeviceType getLibtorchDevice() const { return _libtorch_device; }                                                
#endif                                                                                                                    
                                                                                                                          
  /**                                                                                                                     
   * Stores configuration options relating to the fixed-point solving                                                     
   * capability.  This is used for communicating input-file-based config from                                             
   * the MultiApp object/syntax to the execution (e.g. executor) system.                                                  
   */                                                                                                                     
  struct FixedPointConfig
```
e.g. the `/** */` and `///` syntax

Reply:
It's already pushed with this added, got he info from @permcody  in the meantime of what you are referring to.